### ISSUE TITLE
Wrong Loss function in MNIST-ACGAN example

### ISSUE THREAD

**Issue by npmhung** [keras-team/keras#8997]
The original [paper](https://arxiv.org/pdf/1610.09585.pdf) of AC-Gan claims that it optimizes the Generator by maximizing the L_c - L_s, different from the function, which is L_c + L_s, to optimize the Discriminator by maximizing that function. 

However, in the [source code](https://github.com/keras-team/keras/blob/master/examples/mnist_acgan.py) of AC-Gan in Keras repository, the author optimizes both Generator and Discriminator by maximizing L_c + L_s. 

Is that a mistake or are they doing this on purpose?

**Comment by ozabluda** [comment]
In the paper `L_s` is the same for the Discriminator and Generator `L_s = ln(1-p)`, while in the example `L_s` for the generator is different: `L_s = -ln(p)`, which works better.
